packages/bus/src/Brokers/RMQEventBus.ts

```diff
@@ -0,0 +1,395 @@
+import { ExponentialBackoffRetryStrategy } from '../RetryStrategies';
+import { ConnectionFactory } from '../Factories/ConnectionFactory';
+import {
+  Command,
+  Event,
+  EventBus,
+  EventHandler,
+  NoResponse,
+  RetryStrategy,
+  Configuration,
+  Credentials,
+  IllegalOperation,
+  EventHandlerNotFound,
+  NoSubscriptionsFound,
+  EventHandlerConstructor
+} from '@secbox/core';
+import { ConfirmChannel, Connection, ConsumeMessage } from 'amqplib';
+import { DependencyContainer, inject, injectable } from 'tsyringe';
+import { EventEmitter, once } from 'events';
+import { format, parse, UrlWithParsedQuery } from 'url';
+
+interface ParsedConsumeMessage<T = unknown> {
+  payload: Event<T>;
+  name: string;
+  replyTo?: string;
+  correlationId?: string;
+}
+
+@injectable()
+export class RMQEventBus implements EventBus {
+  private client?: Connection;
+  private channel?: ConfirmChannel;
+
+  private readonly subject = new EventEmitter({ captureRejections: true });
+  private readonly handlers = new Map<
+    string,
+    EventHandler<unknown, unknown>[]
+  >();
+  private readonly consumerTags: string[] = [];
+  private readonly container!: DependencyContainer;
+  private readonly backOffRetry!: RetryStrategy;
+
+  private readonly REPLY_QUEUE_NAME = 'amq.rabbitmq.reply-to';
+  private readonly APP_QUEUE_NAME = 'app';
+  private readonly DEFAULT_RECONNECT_TIMES = 20;
+
+  constructor(
+    @inject(Configuration) private readonly configuration: Configuration,
+    @inject(ConnectionFactory)
+    private readonly connectionFactory: ConnectionFactory<Connection>
+  ) {
+    this.container = configuration.container;
+    this.subject.setMaxListeners(Infinity);
+    this.backOffRetry = new ExponentialBackoffRetryStrategy({
+      maxDepth: this.DEFAULT_RECONNECT_TIMES
+    });
+  }
+
+  public async init(): Promise<void> {
+    const connectTimeout = this.configuration.connectTimeout;
+    const heartbeat = this.configuration.heartbeatInterval;
+    const credentials = this.configuration.credentials;
+    const bus = this.configuration.bus;
+
+    const buildedUrl = this.buildUrl(bus, credentials, heartbeat || 5000);
+
+    this.client = await this.backOffRetry.acquire(async () =>
+      this.connectionFactory.create(buildedUrl, {
+        timeout: connectTimeout
```

Comment:
The preferred way is to use shorthand properties.
```suggestion
        timeout
```